src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,186 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    public static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> previous;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.previous = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (isPositionIndex(index)) {
+            if (index == size) {
+                linkLast(value);
+            } else {
+                linkBefore(value, node(index));
+            }
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null || list.size() == 0) {
+            throw new NullPointerException(" ");
+        } else {
+            for (T t : list) {
+                add(t);
+            }
+        }
+
     }
```

Comment:
You like one-letter variable names, don't you?)
How about using "item" or "value" instead of "t"?